packages/iov-tendermint-rpc/src/v0-31/hasher.spec.ts

```diff
@@ -1,14 +1,59 @@
 import { Encoding } from "@iov/encoding";
+import { ReadonlyDate } from "readonly-date";
 
+import { ReadonlyDateWithNanoseconds } from "../responses";
 import { TxBytes } from "../types";
-import { hashTx } from "./hasher";
+import { hashBlock, hashTx } from "./hasher";
+
+const { fromHex, fromBase64 } = Encoding;
 
 describe("Hasher", () => {
   it("creates transaction hash equal to local test", () => {
     // This was taken from a result from /tx_search of some random test transaction
     // curl "http://localhost:11127/tx_search?query=\"tx.hash='5CB2CF94A1097A4BC19258BC2353C3E76102B6D528458BE45C855DC5563C1DB2'\""
-    const txId = Encoding.fromHex("5CB2CF94A1097A4BC19258BC2353C3E76102B6D528458BE45C855DC5563C1DB2");
-    const txData = Encoding.fromBase64("YUpxZDY2NURaUDMxPWd2TzBPdnNrVWFWYg==") as TxBytes;
+    const txId = fromHex("5CB2CF94A1097A4BC19258BC2353C3E76102B6D528458BE45C855DC5563C1DB2");
+    const txData = fromBase64("YUpxZDY2NURaUDMxPWd2TzBPdnNrVWFWYg==") as TxBytes;
     expect(hashTx(txData)).toEqual(txId);
   });
+
+  it("creates block hash equal to local test for empty block", () => {
+    // This was taken from a result from /block of some random empty block
+    // curl "http://localhost:11131/block"
+    const blockId = fromHex("5B5D3F7E77A4BD6CB6067947E478BC3BD493DD24A981535F0ADEBDAAA0498480");
+    const time = new ReadonlyDate("2019-09-19T10:41:24.898178746Z");
+    // tslint:disable-next-line:no-object-mutation
+    (time as any).nanoseconds = 898178746;
```

Comment:
Untested but makes TS happy: 

```ts
const time: ReadonlyDateWithNanoseconds = {
  ...new ReadonlyDate("2019-09-19T10:41:24.898178746Z"),
  nanoseconds: 898178746,
};
```

Reply:
This doesn't work because you lose the `Date` methods like `getTime`.